### DISCUSSION TITLE
'aws configure get region' does not work with the --profile flag

### DISCUSSION THREAD

**Discussion by YevheniiPokhvalii** [aws/aws-cli#7016] [Q&A] [answered]
### Describe the bug

When I run `aws configure get region` it only gets the region for the `default` profile. When I try to add `--profile` to this command, it returns nothing.

```
$ aws configure get region
eu-central-1
$ aws configure get region --profile default
eu-central-1
```
```
$ aws configure get region --profile some_other_profile
$ 
$ # empty result
```

### Expected Behavior

```
$ aws configure get region --profile some_other_profile
eu-central-1
```

`aws configure get region --profile some_other_profile` should return the default region for the chosen profile.

### Current Behavior

```
$ aws configure get region --profile some_other_profile
$ 
$ # empty result
```

### Reproduction Steps

Run `aws configure get region --profile some_other_profile`. This profile should not be `default`.

### CLI version used

aws-cli/2.7.3

### Environment details (OS name and version, etc.)

Ubuntu 22.04 (WSL) Windows 11

**Comment by kdaily** [comment, marked as answer]
Hi @YevheniiPokhvalii,

Sorry to hear you're having an issue. I was not able to reproduce this - providing a `--profile` value returns the region set for that profile. I have in my `~/.aws/config` file:

```
[profile some_other_profile]
region=ap-south-1
```

Running this:

```
$ aws configure get region --profile some_other_profile
ap-south-1
```

What do you get if you run `aws configure list --profile some_other_profile`?

You can also check what the debug logs show by adding `--debug` to your command. You should see (among other statements) the following:

```
...
2022-06-01 09:36:08,529 - MainThread - botocore.session - DEBUG - Setting config variable for profile to 'some_other_profile'
...
2022-06-01 09:36:08,541 - MainThread - awscli.customizations.configure.get - DEBUG - Config value retrieved: ap-south-1
```

**Comment by YevheniiPokhvalii** [comment]
@kdaily Hey, I realised what the issue was:

I have the following naming in my `~/.aws/config`:
```
[some-profile]
region = eu-central-2
output = json
```

And this way, `aws configure get region --profile some-profile` does not work. The profile itself should be changed to:
```
[profile some-profile]
region = eu-central-2
output = json
```

Is the `profile` prefix mandatory now in the `~/.aws/config` file? 
It seems that `aws configure get region --profile some-profile`  does not work with profiles with old naming convention but works with the `default` profile that does not have that `profile` prefix.

**Comment by kdaily** [comment]
Hi @YevheniiPokhvalii,

Glad it's working now! The `profile` prefix has been mandatory in the config file for a long time; the default profile is the only exception. You can read more about the configuration file format and specification here:

https://docs.aws.amazon.com/sdkref/latest/guide/file-format.html#file-format-config

I'm going to convert this to a discussion forum post because it is a guidance question and will likely come up again! Please mark it as answered if you agree with the answer provided. Thanks!

**Reply by YevheniiPokhvalii** [reply]
Thanks! I noticed that many people from my team copy-paste their ancient AWS config manually from computer to computer and that's why this issue occurs.
Are there any difficulties to implement the ability for this command to read a region with the old profile naming?

And another question:
Should `$AWS_DEFAULT_REGION` or `$AWS_REGION` be picked up by `aws configure get region` ?
From what I see, this command only reads the region from `~/.aws/config`.
